slep005/proposal.rst

```diff
@@ -177,10 +177,15 @@ transformation within a ``FeatureUnion`` or ``ColumnTransformer``. Thus the
 .. rubric:: Handling ``fit`` parameters
 
 Sample props or weights cannot be routed to steps downstream of a resampler in
-a ``Pipeline``, unless they too are resampled. It's very unclear how this would
-work with ``Pipeline``'s current prefix-based fit parameter routing.
-
-TODO: propose solutions
+a Pipeline, unless they too are resampled. To support this, a resampler
+would need to be passed all props that are required downstream, and
+``fit_resample`` should return resampled versions of them. Note that these
+must be distinct from parameters that affect the resampler's fitting.
+That is, consider the signature ``fit_resample(X, y=None, props=None, sample_weight=None)``.
+The ``sample_weight`` passed in should affect the resampling, but does not
```

Comment:
So if I understand correctly, one could apply a set of sample to drive the resampling and pass another set of weight which will be resampled to later be used by the downstream estimators?

Did you encounter any use case for `sample_weight` in practice? I mean that introducing this parameter might make the sampling API more difficult to grasp and if there is no use case maybe we are better to not allow it.

Note: I am not against including this parameter, this is more by a lack of knowledge regarding the use-case.

Reply:
Let me merge the PR and move the comment to the main repo